README.md

```diff
@@ -286,22 +286,64 @@ To run everything, run:
 make check
 ```
 
-To individually run the unit tests, run `make test` to run the suite in parallel (fast), or run `make test-random` to run the tests in random order (slower, but this is what `make check` runs and what runs in CI).
+To individually run the unit tests, run `make test` to run the suite in parallel (fast), or run
+`make test-random` to run the tests in random order (slower, but this is what `make check` runs and
+what runs in CI).
 
 ### Functional Tests
 
-Functional tests are run alone using `make test-functional` and otherwise will be ran along with `make check`.
+Functional tests are run alone using `make test-functional` and otherwise will be ran along with
+`make check`.
 
-Some of the tests appear to get into a state that reliably causes subsequent tests to crash. Such tests have been isolated and are clearly marked. The Makefile is used to ensure we exercise them in a completely new process.
-Use the `qtbot` object to drive the UI. This is part of the [pytest-qt](https://pytest-qt.readthedocs.io/en/latest/) package.
+We use `qtbot`, bundled with the [pytest-qt](https://pytest-qt.readthedocs.io/en/latest/) package,
+for UI interaction within our functional tests. We use
+[vcrpy](https://vcrpy.readthedocs.io/en/latest/) to replay the original response from the test
+server. These responses are stored in the `tests/functional/cassettes` directory. Our test TOTP is
+set to `994892` and stored in casettes.
 
-When writing tests that require the user to log in, on first run of the test
-you must make sure the TOTP value in `conftest.py` is correct for the time at which the test is run.
-For any further run of the test, this doesn't need to be the case since [vcrpy](https://vcrpy.readthedocs.io/en/latest/)
-will replay the original response from the test server. These responses are
-stored in the cassettes directory and should be committed to the git
-repository. Before committing, set the TOTP value in the cassette back to the value we use across all functional tests: `994892`.
+#### Generating new cassettes
 
+Some changes may require generating new cassettes, such as modifications to API endpoints. If you
+see the following warning, you may need to generate a new cassette or all new cassettes, depending
+on the change you made:
+
+```
+Can't overwrite existing cassette ('<path-to-cassette-for-a-functional-test>') in your
+current record mode ('once').
+```
+
+To generate new cassettes, follow these instructions:
+
+0. Before generating new cassettes that will be used to replay server interaction, bypass TOTP
+   verification so that we can use the hard-coded value of `994892` in `tests/conftest.py` by
```

Comment:
making this `123456` would make it clearer that it is a test value and that we are skipping totp generation - ill update this in the conftest.py file and regenerate the cassettes